### ISSUE TITLE
[Android] Modify firebase remote notification

### ISSUE THREAD

**Issue by Etienne-M** [notifee/react-native-notifee#183]
I'm wondering if there's a way to use `notifee` to modify a notification displayed on the device that was sent by firebase.

To give more context: in our android application, we had the following flow:
1.  Our backend would send us a data-only notification using firebase;
2. We would combine the data in the notification with some client-side only data (namely, contact matching);
3. We would then display a local notification using `notifee` with the combined backend and client data.

This flow isn't ideal because data-only notifications do not have the same priorities. I was wondering if there was a way to receive a remote notification, and then edit the displayed notification by updating its content with our client-side data in the handler registered using firebase messaging.

I've tried the naive approach of simply grabbing the id given with the firebase notification and plopping it in `notifee.displayNotification` as described [here](https://notifee.app/react-native/docs/displaying-a-notification#updating-a-notification). I was hoping that doing so would update the content of the notification. This did not work.

My question is two-fold:
1. Is it feasible to edit a notification that was displayed by receiving it from Firebase using plain-old Java/Kotlin native code?
2. Is there a way to do it using `notifee`'s API?

**Comment by helenaford** [comment]
Hi, the docs is referring to updating a notification created by notifee only. Maybe there's something we can add to make the docs more clear. 

It would have been nice to be able to update firebase messages that way.

For iOS, there will be an option to add notifee properties to the FCM payload which is coming soon. As of Android, at this time I don't believe there's a way to do so (not that I know of).

Have you taken a look at our [Power Manager APIs ](https://notifee.app/react-native/docs/android/background-restrictions)? They're helper endpoints to assist your app users to configure their devices to help stop the app from being killed in the background.

**Comment by daralthus** [comment]
In my experience sending data only notifications like this 
```
{
  priority: "high",
  contentAvailable: true,
  mutableContent: true,
  data: { ... }
  // note that there is no "notification" key
}
```
reliably calls the handler set by`firebase.messaging().setBackgroundMessageHandler` on both platforms (even after force quit). Then you can "upgrade" and show your messages with Notifee.

@helenaford Do you know, does this approach have any setbacks that I don't see yet?

**Comment by mikehardy** [comment]
This is exactly what I do in my work project, though I wasn't aware that even on iOS it would work after force quit - I had thought we were stuck in that case, but the results are inconsistent. Either way if a user force quit my app I can't complain too much, I shouldn't have made them want to do that :-).

**Comment by daralthus** [comment]
Maybe worth adding that you need the background notification capability enabled
![image](https://user-images.githubusercontent.com/57560/99813296-686f7100-2b3f-11eb-9729-43f7a11512c5.png)

**Comment by mikehardy** [comment]
Yes definitly - but that is react-native-firebase / FCM-related I think? The docs should be (I hope) clear over there, but from a notifee perspective I don't think it in scope over here?

**Comment by helenaford** [comment]
@daralthus that's what I've done too in my own apps and it works great.

@mikehardy yea we could add a note to the fcm integration section (https://notifee.app/react-native/docs/integrations/fcm#integrating-notifee) about setting the priority if that'll be helpful?

**Comment by helenaford** [comment]
@Etienne-M  Just wanted to post a quick update where we are on this. For iOS, it's now possible to send a firebase message and to allow notifee to modify the notification before it's displayed to the end-user via a Notification Service Extension.  Please see here for more details (https://notifee.app/react-native/docs/ios/remote-notification-support). This will help if you are facing any background issues. For Android, we're looking into if we can do something similar but the most reliable way, currently, is by sending data-only messages as we previously discussed above.

**Comment by alloystory** [comment]
@helenaford hello! just wondering if this is still being looked into?

**Comment by helenaford** [comment]
@alloystory I opened up an issue on the Firebase Android SDK repo (https://github.com/firebase/firebase-android-sdk/issues/2639). I looked into it since my last comment and found out it's not possible until it's been updated by the native SDK. For now, the only way to do it is data-only notifications.
